### ISSUE TITLE
Supporting Spark MapType in SchemaConverters

### ISSUE THREAD

**Issue by BenMizrahiPlarium** [GoogleCloudDataproc/spark-bigquery-connector#522]
Using schema convertor throws an exception when using **spark MapType** because of the following code:  
 
https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/1221996f88b54970f8f3d3ed2d68b72dfd9d3eee/spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java#L435

While in BigQuery it is possible to use MapType using ORC loader:

![image](https://user-images.githubusercontent.com/24546394/149661190-4b1d71ab-ee05-4df9-9f59-6b5fb3212e2f.png)

I did local implementation of SchemaConverters with the following support of spark MapType:

``` 
....
    private static FieldList sparkToBigQueryMapFields(MapType sparkMap) {
        ArrayList<Field> bqFields = new ArrayList<Field>();
        DataType key =  sparkMap.keyType();
        Field.Builder fieldBuilderKey = createBigQueryFieldBuilder("key", toBigQueryType(key), Field.Mode.NULLABLE, null);
        bqFields.add(fieldBuilderKey.build());

        DataType value =  sparkMap.valueType();
        Field.Builder fieldBuilderValue = createBigQueryFieldBuilder("value", toBigQueryType(value), Field.Mode.NULLABLE, null);
        bqFields.add(fieldBuilderValue.build());

        return FieldList.of(bqFields);
    }


protected static Field createBigQueryColumn(StructField sparkField, int depth) {
        DataType sparkType = sparkField.dataType();
        String fieldName = sparkField.name();
        Field.Mode fieldMode = sparkField.nullable() ? Field.Mode.NULLABLE : Field.Mode.REQUIRED;
        FieldList subFields = null;
        if (sparkType instanceof ArrayType) {
            ArrayType arrayType = (ArrayType)sparkType;
            fieldMode = Field.Mode.REPEATED;
            sparkType = arrayType.elementType();
        }
        LegacySQLTypeName fieldType;

        if(sparkType instanceof MapType){
            fieldMode = Field.Mode.REPEATED;
            fieldType = LegacySQLTypeName.RECORD;
            subFields = sparkToBigQueryMapFields((MapType)sparkType);
        }
        else if (sparkType instanceof StructType) {
            subFields = sparkToBigQueryFields((StructType)sparkType, depth + 1);
            fieldType = LegacySQLTypeName.RECORD;
        } else {
            fieldType = toBigQueryType(sparkType);
        }

        Field.Builder fieldBuilder = createBigQueryFieldBuilder(fieldName, fieldType, fieldMode, subFields);
        Optional<String> description = getDescriptionOrCommentOfField(sparkField);
        if (description.isPresent()) {
            fieldBuilder.setDescription((String)description.get());
        }

        return fieldBuilder.build();
    }

....
```

**Comment by davidrabinowitz** [comment]
Added in PRs #914 and #917. Will be released in version 0.30

**Comment by BenMizrahiPlarium** [comment]
Awesome thank you @davidrabinowitz !
Looking forward to test
